lib/quicly.c

```diff
@@ -284,9 +319,43 @@ struct st_quicly_conn_t {
          */
         int64_t send_ack_at;
         /**
-         *
+         * congestion control
          */
         quicly_cc_t cc;
+        /**
+         * pending NEW_CONNECTION_ID
+         */
+        struct {
+            /**
+             * maximum number of path ID issued so far
+             */
+            uint32_t last_path_id;
+            /**
+             * list of struct st_quicly_pending_new_cid_t
+             */
+            struct st_quicly_pending_new_cid_t *cids;
```

Comment:
Hmm, are you introducing a new quicly-internal limit `N`, which is basically independent from `MAX_PATH_ID` or `peer.tp.active_connection_id_limit` (*)?

If that's the case I think it makes sense. In case the peer announced `peer.tp.active_connection_id_limit = 255`, it sounds too much to issue 254 CIDs at once (which is what my code currently does.)

(*) given `N <= min(MAX_PATH_ID, peer.tp.active_connection_id_limit)` is satisfied

Reply:
What you have stated is exactly what I meant.

And the limit can be equal to local.tp.active_connection_id_limit. This is because client-issued CIDs and server-issued CIDs are typically used in pair.